gmso/utils/misc.py

```diff
@@ -5,6 +5,19 @@
 from unyt.exceptions import UnitConversionError
 
 
+@lru_cache(maxsize=128)
+def unyt_compare(units1, units2):
+    """Check if two parameter values have the same units."""
+    units1 = list(units1)
+    units2 = list(units2)
+    for unit1, unit2 in zip(units1, units2):
+        try:
+            u.testing.assert_allclose_units(unit1, unit2, rtol=1e-5, atol=1e-8)
+            return True
+        except AssertionError:
+            return False
```

Comment:
```suggestion
    for unit1, unit2 in zip(units1, units2):
        try:
            u.testing.assert_allclose_units(unit1, unit2, rtol=1e-5, atol=1e-8)
        except AssertionError:
            return False
    return True
```

Reply:
This works just as well I believe, although I think it still runs into the fundamental issue that looking for an AssertionError isn't exactly what we want to do here.